handlers/categoryHandler.js

```diff
@@ -0,0 +1,52 @@
+const express = require("express");
+const router = express.Router();
+const { db } = require("../firebase");
+
+router.get("/", (req, res) => {
+  const category = db.collection("Category");
+  category
+    .get()
+    .then((snapshot) => {
+      const data = snapshot.docs.map((doc) => ({
+        id: doc.id,
+        name: doc.get("name"),
+      }));
+      console.log(data);
+      res.status(200);
+      res.send(data);
+    })
+    .catch((error) => {
+      console.error(error);
+      res.status(500);
+      res.send("Something went wrong, please try again");
+    });
+});
+
+router.post('/', (req, res) => {
+  let name = req.body.name;
+  let sponser = req.body.sponser;
+  let audience_size = req.body.audience_size;
+  let duration = req.body.duration;
+
+  let message = "";
+
+  if (name != "" && sponser != null && audience_size != "" && duration != "") {
```

Comment:
why sponser is compared with null and others with empty string?

Reply:
coz i thought true false is not supposed to be string

Reply:
but your variable will always collect string data type from request.. have you typecasted to any specific datatype?